### PR TITLE
Pdct 312 Added endpoint for creating events

### PR DESCRIPTION
**Pull request by katybaulch**
# Description

Please include:
- Added endpoint for creating events.
- Added service and repo functionality for creating tests.
- Added tests for create endpoint.

[Link to Linear ticket](https://linear.app/climate-policy-radar/issue/PDCT-312/endpoint-for-creating-events)

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## How Has This Been Tested?

Unit and integration tests have been added.

## Reviewer Checklist

- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: integration_tests/mocks/rollback_event_repo.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    # def mock_update_document(db, import_id: str, data: EventReadDTO) -> EventReadDTO:
+    #     actual_update(db, import_id, data)
+    #     raise NoResultFound()
+
+    def mock_create_document(db, data: EventCreateDTO) -> Optional[EventReadDTO]:
+        actual_create(db, data)
+        raise NoResultFound()
+
+    # def mock_delete_document(db, import_id: str) -> bool:
```
Comment (diversemix) [inline integration_tests/mocks/rollback_event_repo.py:22]: Nice pre-emptive coding :wink:

File: app/service/event.py [hunk 2]
```diff
@@ -73,6 +75,27 @@ def validate_import_id(import_id: str) -> None:
...
+        :raises RepositoryError: raised on a database error
+        :raises ValidationError: raised should the import_id be invalid.
+        :return Optional[documentDTO]: The new created document or
+    None if unsuccessful.
+    """
+    id.validate(event.family_import_id)
+
+    family = family_service.get(event.family_import_id)
+    if family is None:
+        raise ValidationError(f"Could not find event for {event.family_import_id}")
```
[the remark this answers is not in the document]
Comment (katybaulch) [reply app/service/event.py:94]: Sorted.

File: integration_tests/mocks/rollback_event_repo.py (hunk 1 above)
Comment (katybaulch) [reply integration_tests/mocks/rollback_event_repo.py]: Whoops - great spot. Have removed all the commented out code in this file.

File: integration_tests/setup_db.py [hunk 3]
```diff
@@ -362,6 +357,7 @@ def _setup_event_data(
             date=data["date"],
             event_type_name=data["event_type_value"],
             family_import_id=data["family_import_id"],
+            # family_document_import_id=data["family_document_import_id"],
```
Comment (diversemix) [inline integration_tests/setup_db.py:360]: Would be nice to have a test that checked this is set

**Review by diversemix — APPROVED** [review]
Pre-emptive approve when comment is back :100: great stuff

File: unit_tests/routers/test_event.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 from fastapi import status
+from fastapi.encoders import jsonable_encoder
```
Comment (diversemix) [inline unit_tests/routers/test_event.py:2]: Ooo nice - didn't know about these

Comment (katybaulch) [reply unit_tests/routers/test_event.py:2]: It was a necessity - if you pass a Pydantic model to FastAPI via JSON with data types that aren't inherently serialisable to JSON such as Python datetime, the tests will fail. Fun nugget of info for this afternoon!

File: integration_tests/setup_db.py (hunk 3 above)
Comment (katybaulch) [reply integration_tests/setup_db.py]: Have readded the todo that seemed to get lost during the merge.

**Pull request merged**